packages/airnode-node/src/evm/handlers/process-transactions.test.ts

```diff
@@ -18,114 +19,184 @@ mockEthers({
+      const gasPriceSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getGasPrice');
+      const blockSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getBlock');
+      if (txType === '1') {
+        gasPriceSpy.mockRejectedValue(new Error('Gas price cannot be fetched'));
+      } else {
+        blockSpy.mockRejectedValue(new Error('Block header cannot be fetched'));
       }
-    );
-
-    // API call was submitted
-    expect(fulfillMock).toHaveBeenCalledTimes(1);
-    expect(fulfillMock).toHaveBeenCalledWith(
-      apiCall.id,
-      apiCall.airnodeAddress,
-      apiCall.fulfillAddress,
-      apiCall.fulfillFunctionId,
-      '0x000000000000000000000000000000000000000000000000000000000001252b',
-      '0x34c1f1547c1f2f7c3a8bd893e20444ccee56622d37a18b7dc461fb2359ef044e3b63c21e18a93354569207c7d21d1f92f8e8a310a78eeb9a57c455052695491f1b',
-      { gasLimit: 500_000, gasPrice, nonce: 79 }
-    );
-  });
-
-  it('does not submit transactions if a gas price cannot be fetched', async () => {
-    const contract = new ethers.Contract('address', ['ABI']);
-    const gasPriceSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getGasPrice');
-    gasPriceSpy.mockRejectedValue(new Error('Gas price cannot be fetched'));
-
-    const apiCall = fixtures.requests.buildSubmittableApiCall({
-      sponsorAddress: '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181',
-    });
-    const requests: GroupedRequests = {
-      apiCalls: [apiCall],
-      withdrawals: [],
-    };
-    const transactionCountsBySponsorAddress = { '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181': 79 };
-    const state = fixtures.buildEVMProviderState({
-      requests,
-      transactionCountsBySponsorAddress,
-    });
-
-    const res = await processTransactions(state);
-    expect(res.requests.apiCalls[0]).toEqual({ ...apiCall, nonce: 79 });
-    expect(res.gasTarget).toEqual(null);
-
-    // API call was NOT submitted
-    expect(contract.fulfill).not.toHaveBeenCalled();
-  });
+
+      const apiCall = fixtures.requests.buildSubmittableApiCall({
+        sponsorAddress: '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181',
+      });
+      const requests: GroupedRequests = {
+        apiCalls: [apiCall],
+        withdrawals: [],
+      };
+      const transactionCountsBySponsorAddress = { '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181': 79 };
+      const initialState = fixtures.buildEVMProviderState({
+        requests,
+        transactionCountsBySponsorAddress,
+      });
+
+      const chainOptions = { txType } as ChainOptions;
+
+      const state = {
+        ...initialState,
+        settings: {
+          ...initialState.settings,
+          chainOptions,
+        },
+      };
+
+      const res = await processTransactions(state);
+      expect(res.requests.apiCalls[0]).toEqual({ ...apiCall, nonce: 79 });
+      expect(res.gasTarget).toEqual(null);
+
+      // API call was NOT submitted
+      expect(contract.fulfill).not.toHaveBeenCalled();
```

Comment:
This `contract` is just an instance you created. It is not used anywhere so it will never be submitted. You can mock one of the Airnode internal functions (e.g. `submit`) but I'd like to see it being called in one test and NOT being called in another - but I think it's not necessary.

Reply:
I don't remember writing this, I think it was part of the test originally, but I will work on it.

Reply:
This should be fixed now, replaced with a spy.